api/src/main/java/io/kafbat/ui/service/TopicsService.java

```diff
@@ -320,28 +333,35 @@ private Map<TopicPartition, Optional<NewPartitionReassignment>> getPartitionsRea
         var partition = assignmentEntry.getKey();
         var brokers = assignmentEntry.getValue();
 
+        // Copy from online nodes if all nodes are offline
+        if (brokers.isEmpty()) {
+          brokers = new ArrayList<>(onlineNodes);
+        }
+
         // Get brokers list sorted by usage in reverse order
         var brokersUsageList = brokersUsage.entrySet().stream()
             .sorted(Map.Entry.comparingByValue(Comparator.reverseOrder()))
             .map(Map.Entry::getKey)
             .toList();
 
+        Integer leader = topic.getPartitions().get(partition).getLeader();
+
         // Iterate brokers and try to remove them from assignment
         // while partition replicas count != requested replication factor
         for (Integer broker : brokersUsageList) {
+          if (brokers.size() == replicationFactorChange.getTotalReplicationFactor()) {
+            break;
+          }
           // Check is the broker the leader of partition
-          if (!topic.getPartitions().get(partition).getLeader()
-              .equals(broker)) {
+          if (leader == null || !leader.equals(broker)) {
             brokers.remove(broker);
             brokersUsage.merge(broker, -1, Integer::sum);
```

Comment:
why we doin `brokersUsage.merge(broker, -1, Integer::sum);` here and other places? looks like it is not used)

Reply:
We are trying to keep leader if it is presented